python/lsst/afw/math/stack.cc

```diff
@@ -32,6 +35,31 @@ using namespace py::literals;
 
 using namespace lsst::afw::math;
 
+namespace {
+
+// a variant of statisticsStack for pybind11 that takes vector of vector
+// (instead of vector of shared pointer to vector)
+// and returns a vector instead of a shared pointer to a vector
+template <typename PixelT>
+std::vector<PixelT> _statisticsStack(
+        std::vector<std::vector<PixelT>>& vectors,  ///< Vectors to process
+        Property flags,                                              ///< statistics requested
+        StatisticsControl const& sctrl = StatisticsControl(),        ///< control structure
+        std::vector<lsst::afw::image::VariancePixel> const& wvector =
+                std::vector<lsst::afw::image::VariancePixel>(0)  ///< vector containing weights
+) {
+        std::vector<std::shared_ptr<std::vector<PixelT>>> vcopy;
+        vcopy.reserve(vectors.size());
+        for (auto const & v : vectors) {
+            auto vshared = std::make_shared<std::vector<PixelT>>(std::move(v));
+            vcopy.emplace_back(vshared);
+        }
+        auto result = statisticsStack(vcopy, flags, sctrl, wvector);
+        return std::vector<PixelT>(result->begin(), result->end());
```

Comment:
This seems unnecessarily convoluted. Why not just `vector<PixelT>(*result)`? If anything, a constructor that knows it's dealing with a vector may be more optimal than one that works on a generic iterator.

Reply:
Or `return std::vector<PixelT>(*statisticsStack(vcopy, flags, sctrl, wvector))` to make sure the compiler knows it is a temporary enabling a move? It may or may not do that now too though.